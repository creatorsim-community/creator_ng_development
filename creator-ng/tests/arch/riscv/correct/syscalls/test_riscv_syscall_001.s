#
# Creator (https://creatorsim.github.io/creator/)
#

.data

.align 2

  byte:			.byte 120,255
  half:			.half 34
  word:			.word -5678
  string:  		.string "This is a string"

.align 2
  float:     .float 1.1

.text
main:
	# print int
	li a7, 1
	li a0, 6673
	ecall

	# print byte
  la t1, byte
  lb a0, 0(t1)
  ecall
  
  # print half
  la t1, half
  lh a0, 0(t1)
  ecall
  
  # print negative int
  la t1, word
  lw a0, 0(t1)
  ecall
  
  # print float
  la t1, float
  flw f0, 0(t1)
  fmv.x.w a0, f0 # convert float to int to print it. This will print 1066192077, or 0x3F8CCCCD in hex
  ecall
  
  # string string
  li a7, 4
  la a0, string
  ecall
  jr ra