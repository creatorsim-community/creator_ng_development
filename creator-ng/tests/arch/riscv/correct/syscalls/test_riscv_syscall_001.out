
6673
120
34
-5678
1066192077
this is a string
cr[PC]:0x64; ir[x6,t1]:0x20001c; ir[x10,a0]:0x200008; ir[x17,a7]:0x4; fpr[f0,ft0]:0xffffffff3F8CCCCD; keyboard[0x0]:''; display[0x0]:'667312034-56781066192077this%20is%20a%20string'; 

